src/main/java/net/sf/jabref/gui/worker/CitationStyleToClipboardWorker.java

```diff
@@ -0,0 +1,195 @@
+package net.sf.jabref.gui.worker;
+
+import java.awt.datatransfer.StringSelection;
+import java.awt.datatransfer.Transferable;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Pattern;
+
+import javax.swing.SwingWorker;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.gui.BasePanel;
+import net.sf.jabref.gui.ClipBoardManager;
+import net.sf.jabref.gui.exporter.RtfTransferable;
+import net.sf.jabref.gui.fieldeditors.HtmlTransferable;
+import net.sf.jabref.gui.fieldeditors.XmlTransferable;
+import net.sf.jabref.logic.citationstyle.CitationStyle;
+import net.sf.jabref.logic.citationstyle.CitationStyleGenerator;
+import net.sf.jabref.logic.citationstyle.CitationStyleOutputFormat;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.layout.Layout;
+import net.sf.jabref.logic.layout.LayoutFormatterPreferences;
+import net.sf.jabref.logic.layout.LayoutHelper;
+import net.sf.jabref.logic.util.OS;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.preferences.PreviewPreferences;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Copies the selected entries and formats them with the selected citation style (or preview), then it is copied to the clipboard.
+ * This worker cannot be reused.
+ */
+public class CitationStyleToClipboardWorker extends SwingWorker<List<String>, Void> {
+
+    private static final Log LOGGER = LogFactory.getLog(CitationStyleToClipboardWorker.class);
+    private static final Pattern REMOVE_HTML = Pattern.compile("<(?!br)(?!BR).*?>");
+    private static final Pattern WHITESPACE = Pattern.compile("(?m)^\\s|\\v+");
+    private static final Pattern HTML_NEWLINE = Pattern.compile("<br>|<BR>");
+
+    private final BasePanel basePanel;
+    private final List<BibEntry> selectedEntries;
+    private final String style;
+    private final String previewStyle;
+    private final CitationStyleOutputFormat outputFormat;
+
+
+    public CitationStyleToClipboardWorker(BasePanel basePanel, CitationStyleOutputFormat outputFormat) {
+        this.basePanel = basePanel;
+        this.selectedEntries = basePanel.getSelectedEntries();
+        PreviewPreferences previewPreferences = Globals.prefs.getPreviewPreferences();
+        this.style = previewPreferences.getPreviewCycle().get(previewPreferences.getPreviewCyclePosition());
+        this.previewStyle = Globals.prefs.getPreviewPreferences().getPreviewStyle();
+        this.outputFormat = outputFormat;
+
+        basePanel.frame().setStatus(Localization.lang("Copying..."));
+    }
+
+    @Override
+    protected List<String> doInBackground() throws Exception {
+        if (CitationStyle.isCitationStyleFile(style)) {
```

Comment:
This part looks rather generic and might be reusable (I expect similar code for the generation of the entry preview).
I'm not sure if this makes sense, but maybe we should introduce a new class on top of the citation style generator and our Layout class. I just want to tell somebody: here is a bunch of BibEntries, please give me a nice string representation in a format I choose; but I don't care how you generate this - citation style, layout or simple string manipulation.
But this is probably something for a new PR.